nemo/collections/nlp/models/language_modeling/megatron_lm_encoder_decoder_model.py

```diff
@@ -1223,14 +1244,17 @@ def dummy():
                         dim=-1, index=torch.tensor(ignore_ids, device=device), value=-float('Inf')
                     )
 
-                # TODO: do log_softmax in fp32?
-                log_probs, token_ids = torch.max(torch.nn.functional.log_softmax(output_tensor, dim=-1), dim=-1)
+                log_probs, token_ids = sample_token_fn(logits=output_tensor[:, -1, :])
+                # enforce valid range of token ids
+                token_ids = torch.clamp(token_ids, max=tokenizer.vocab_size - 1)
```

Comment:
Not related to this PR, but this seems like a weird use of clamp. Should a token_ids > vocab raise an error? @ericharper

Reply:
It can happen due to vocabulary padding (mostly in the early stages of training).